### ISSUE TITLE
Feature request : option to use the form id instead of concatenated label ?

### ISSUE THREAD

**Issue by jy95** [issue-ops/parser#72]
Hello @ncalteen,

First things first, thanks for your hard work this action : quite useful indeed.
I created a couple of [YAML Issue Template](https://github.com/jy95/jy95.github.io/tree/master/.github/ISSUE_TEMPLATE) and was wondering if it was possible to use the id of template instead of the label as key, by some kind of option ? As an option, to preserve your backwards compatibility ofc.

## Example of the to-be

```
steps:
  - name: Parse Issue
    id: parser
    uses: issue-ops/parser@vX.X.X
    with:
      body: ${{ github.event.issue.body }}
      useIdsFromForm: true

  - name: Output Issue JSON
    id: output-issue
    run: echo ${{ steps.parser.outputs.json }}
```

Taking your [example issue template](https://github.com/issue-ops/parser/blob/main/__fixtures__/example/template.yml) and your issue example : 

```md
### The Name of the Thing

this-thing

### The Nickname of the Thing

thing

### The Color of the Thing

blue

### The Shape of the Thing

square

### The Sounds of the Thing

re, mi

### The Topics About the Thing

_No response_

### The Description of the Thing

This is a description.

It has lines.

### The Notes About the Thing

- Note
- Another note
- Lots of notes

### The Code of the Thing

const thing = new Thing()

thing.doThing()

### The String Method of the Code of the Thing

thing.toString()

### Is the Thing a Thing?

- [x] Yes
- [x] No

### Is the Thing Useful?

- [ ] Yes
- [x] Sometimes
- [ ] No

### Read Team

IssueOps-Demo-Readers

### Write Team

IssueOps-Demo-Writers-NotATeam

```

We will get : 

```json
{
  "name": "this-thing",
  "nickname": "thing",
  "color": ["blue"],
  "shape": ["square"],
  "sounds": ["re", "mi"],
  "topics": [],
  "description": "This is a description.\n\nIt has lines.",
  "notes": "- Note\n- Another note\n- Lots of notes",
  "code": "const thing = new Thing()\n\nthing.doThing()",
  "code-string": "thing.toString()",
  "is-thing": {
    "selected": ["Yes"],
    "unselected": ["No"]
  },
  "is-thing-useful": {
    "selected": ["Sometimes"],
    "unselected": ["Yes", "No"]
  },
  "read-team": "IssueOps-Demo-Readers",
  "write-team": "IssueOps-Demo-Writers"
}
```

Thanks in advance

**Comment by jy95** [comment]
For the being time, I wrote a [Github actions workaround](https://github.com/jy95/jy95.github.io/blob/master/.github/workflows/issuesAutomatedTasks.yml) that I don't find optimal as it rely on your current implementation of formatKey :

```js
/**
 * Formats a input name to a snake case string
 *
 * - Removes leading and trailing whitespace
 * - Converts to lowercase
 * - Replaces spaces with underscores
 * - Replaces non-alphanumeric characters with underscores
 * - Replaces multiple consecutive underscores with a single underscore
 */
export function formatKey(name: string): string {
  return name
    .trim()
    .toLowerCase()
    .replace(/[^a-z0-9]/g, '_')
    .replace(/^_+|_+$/g, '')
    .replace(/_+/g, '_')
}
```

**Comment by ncalteen** [comment]
Yeah I like this approach a lot more personally. I kinda waffled on it a bit before going with the initial implementation, and looking back using the ID is more consistent IMO. I think what I will do is adjust the behavior to prefer to ID over the actual header text, and then fall back to the header text if no ID is present. That would be potentially less impactful to existing users.

How does that sound?

**Comment by jy95** [comment]
I think it could work as [`id`](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema) is optional by design 👍

> The identifier for the element, except when type is set to markdown. Can only use alpha-numeric characters, -, and _. Must be unique in the form definition. If provided, the id is the canonical identifier for the field in URL query parameter prefills.

**Comment by ncalteen** [comment]
Done! Just released 3.0.0. Give that a try and let me know what you think :)

**Comment by jy95** [comment]
Thanks ncalteen , it seems to work with the first before/after test I did

- PR with 3.0.0 : https://github.com/jy95/jy95.github.io/pull/557
- PR with 2.0.0 : https://github.com/jy95/jy95.github.io/pull/555
